### DISCUSSION TITLE
Error related to repositoriesTask

### DISCUSSION THREAD

**Discussion by lilyevsky** [com-lihaoyi/mill#1276] [Q&A] [answered]
I used to override the "repositories" from CoursierModule trait. Now it says (using mill 0.9.6) that it is deprecated, and I should use repositoriesTask instead. Here is the fragment of my updated build.sc:

```
val lnRepositories = Seq(MavenRepository(localUrl),
    MavenRepository(nexusPublicUrl, Some(Authentication(nexusUsername, nexusPassword)))).
    map(_.asInstanceOf[coursier.Repository])
override def repositoriesTask = T { lnRepositories }
```
It complained about implicit parameter:

> build.sc:29: could not find implicit value for parameter rw: upickle.default.ReadWriter[Seq[coursier.Repository]]
>  override def repositoriesTask = T { lnRepositories }

So then I added the implicit parameter definition (before the repositoriesTask):
```
implicit def rw: upickle.default.ReadWriter[Seq[coursier.Repository]] = upickle.default.macroRW
```
Now I am getting the following error:

> build.sc:27: [error] The referenced trait [[Seq]] must be sealed.
 > implicit def rw: upickle.default.ReadWriter[Seq[coursier.Repository]] = upickle.default.macroRW

What is the solution for such problem?

**Comment by lefou** [comment, marked as answer]
It's probably because you used the `T {}` macros which is supposed to create a [`Target`](https://com-lihaoyi.github.io/mill/mill/Tasks.html#_targets), whereas `def repositoriesTask: Task[Seq[Repository[]]` expects an [anonymous `Task`](https://com-lihaoyi.github.io/mill/mill/Tasks.html#_anonymous_tasks). Just use the `T.task {}` macro, and you should be fine. No upickle reader-writer is needed.

**Reply by lilyevsky** [reply]
Thanks @lefou , that was the exact problem.
